plugins/modules/site_workflow_manager.py

```diff
@@ -698,19 +893,33 @@ def site_requires_update(self):
             specified parameters, such as the site type and site details.
         """
 
-        type = self.have['current_site']['type']
-        updated_site = self.have['current_site']['site'][type]
-        requested_site = self.want['site_params']['site'][type]
+        if self.get_ccc_version_as_integer() <= self.version_2_3_5_3:
+            type = self.have.get('current_site', {}).get('type')
+            updated_site = self.have.get('current_site', {}).get('site', {}).get(type)
+            # requested_site = self.want['site_params']['site'][type]
+            self.log(self.pprint(self.want))
+            requested_site = self.want.get('site_params', {}).get('site', {}).get(type)
+
+        if self.get_ccc_version_as_integer() >= self.version_2_3_7_6:
+            type = self.have.get("current_site").get("type")
+            self.log("Type of site '{0}' for the site creation/updation in Cisco Catalyst Center.".format(type), "DEBUG")
+            current_site = self.have.get('current_site', {})
+            self.log(current_site)
+            updated_site = current_site
+            requested_site = self.want.get('site_params', {}).get('site', {}).get(type)
+
         self.log("Current Site type: {0}".format(str(updated_site)), "INFO")
         self.log("Requested Site type: {0}".format(str(requested_site)), "INFO")
 
         if type == "building":
             return not self.is_building_updated(updated_site, requested_site)
-
         elif type == "floor":
             return not self.is_floor_updated(updated_site, requested_site)
-
-        return not self.is_area_updated(updated_site, requested_site)
+        elif type == "area":
+            return not self.is_area_updated(updated_site, requested_site)
+        else:
+            self.log(f"Unsupported site type '{type}'.", "ERROR")
```

Comment:
self.log(f"Unsupported site type '{0}' given in the playbook. Please provide one of the following site_type: area, building or floor..".format(type), "ERROR")

Reply:
Modified